includes/apple-exporter/components/class-heading.php

```diff
@@ -172,15 +183,15 @@ private static function split_image( $html ) {
 	 * @param string $html The HTML to parse into text for processing.
 	 * @access protected
 	 */
-	protected function build( $html ) {
-		if ( 0 === preg_match( '#<h(\d).*?>(.*?)</h\1>#si', $html, $matches ) ) {
+	protected function build( $html ): void {
+		// Match HTML headings, capture level, id value if set, and heading text.
+		if ( 0 === preg_match( '#<h(\d).*?(id=(["\'])(.*?)\2)?.*?>(.*?)</h\1>#si', $html, $matches ) ) {
 			return;
 		}
-		// TODO add id to identifier for anchor links to headings.
-		// See https://github.com/alleyinteractive/apple-news/pull/1044.
 
 		$level = intval( $matches[1] );
-		$text  = $matches[2];
+		$text  = $matches[5];
+		$id    = $matches[4] ?? null;
```

Comment:
Should there be any checking to insure these array items exist?

Reply:
It's a good question but in the case I don't think it will be necessary because the regex ensures these are always set.